sm/plugin_sm/src/plugin.rs

```diff
@@ -65,13 +86,13 @@ impl ExternalPluginCommand {
         &self,
         action: &str,
         maybe_module: Option<&SoftwareModule>,
-    ) -> Result<Command, SoftwareError> {
+    ) -> Result<LoggedCommand, SoftwareError> {
         let mut command = if let Some(sudo) = &self.sudo {
-            let mut command = Command::new(&sudo);
-            command.arg(&self.path);
+            let mut command = LoggedCommand::new(sudo.to_str().unwrap());
```

Comment:
Is that `unwrap` safe? If `sudo` is always guaranteed to be `Some("sudo")`, then I'm wondering why it's even an `Option` in the first place. Same for the unwraps below.

Reply:
The sudo is optional here because this is not required for the tests.

Reply:
The `sudo` has been made optional to ease testing, and more specifically to work around an issue with tarpaulin and tests using sudo.

Reply:
No the `unwrap()` is not safe. It is currently just because the `sudo` command path is provided internally.

=> Fixed.